yt_dlp/extractor/dplay.py

```diff
@@ -426,3 +436,63 @@ class AnimalPlanetIE(DiscoveryPlusIE):
 
     _PRODUCT = 'apl'
     _API_URL = 'us1-prod-direct.animalplanet.com'
+
+
+class DiscoveryPlusShowBaseIE(InfoExtractor):
+    def _entries(self, show_name):
+        headers = {
+            'x-disco-client': self._X_CLIENT,
+            'x-disco-params': 'realm=' + self._REALM,
+            'referer': self._DOMAIN,
+        }
+        headers['Authorization'] = 'Bearer ' + self._download_json(
+            self._BASE_API + 'token', show_name, 'Downloading token',
+            query={
+                'realm': self._REALM,
+                'deviceId': uuid.uuid4().hex,
+            })['data']['attributes']['token']
```

Comment:
does token have to be re-extracted for each playlist? why not move it into `_real_initialize`?

Reply:
Can you review the headers handling again please?